backend/controllers/userController.js

```diff
@@ -0,0 +1,61 @@
+const asyncHandler = require("express-async-handler");
+const generateToken = require("../utils/jsonUtils");
+const User = require("../models/userModel");
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken")
+
+const registerUser = asyncHandler(async (req, res) => {
+  console.log("HERE");
+  const { name, email, password } = req.body;
+
+  const userExists = await User.findOne({ email });
+
+  if (userExists) {
+    res.status(400); // bad request
+    throw new Error("User already exists");
+  }
+
+  const user = await User.create({
+    name: name,
+    email: email,
+    password: password
+  });
+
+  if (user) {
+    res.status(202).json({
+      _id: user._id,
+      name: user.name,
+      email: user.email,
+      isAdmin: user.isAdmin,
+    });
+  } else {
+    res.status(400);
+    throw new Error("Invalid user data");
+  }
+});
+
+const loginUser = asyncHandler(async (req, res) => {
+  const user = await User.findOne({email: req.body.email});
+  console.log(user);
+  if(!user) {
+      res.status(400);
+      throw new Error("No user found");
+  }
+
+  const validate = await bcrypt.compare(req.body.password, user.password);
+  console.log(validate);
+  if(!validate) {
+      res.status(400);
```

Comment:
I would suggest 401 Unauthorized instead of 400

Reply:
Fixed: Changes are reflected in recent commits